### PR TITLE
fixes for compile errors in EvaluationEngine + small fix to unit test

### PR DESCRIPTION
**Pull request by joaquinvanschoren**
The EvaluationEngine as pulled from master did not compile. This PR fixes the compile errors. It also fixes the unit test on the Number of classes, which now passes.

### CODE REVIEW

File: src/main/java/org/openml/webapplication/algorithm/InstancesHelper.java [hunk 1]
```diff
@@ -125,7 +126,7 @@ else if( total > 0.0 )
 		return result;
 	}
 	
-	public static CostMatrix doubleToCostMatrix(JSONArray cm) {
+	public static CostMatrix doubleToCostMatrix(JSONArray cm) throws JSONException {
```
Comment (janvanrijn) [inline src/main/java/org/openml/webapplication/algorithm/InstancesHelper.java]: why is this necessary? i.e., which line provokes this error.

File: src/test/java/webapplication/TestFantail.java [hunk 2]
```diff
@@ -36,6 +36,6 @@ public void testNumberOfClasses() throws Exception {
 
 		// Check the produced class count
 		Map<String,Double> metafeatures = simpleMetaFeatures.characterize(artificialData);
-		assertSame(metafeatures.get("NumberOfClasses"),0.0);
+		assertEquals(metafeatures.get("NumberOfClasses"),new Double(0.0));
```
Comment (janvanrijn) [inline src/test/java/webapplication/TestFantail.java]: new Double(0.0) is the same as just 0.0, right?

File: src/main/java/org/openml/webapplication/fantail/dc/statistical/SimpleMetaFeatures.java [hunk 3]
```diff
@@ -101,8 +101,8 @@
 
 			// Class properties
 			try {   // If target attribute is defined as numeric, use appropriate values
-				if (dataset.attribute(dataset.classIndex).type() == Attribute.NUMERIC) {
-					NumberOfClasses = 0;
+				if (dataset.attribute(dataset.classIndex()).type() == Attribute.NUMERIC) {
+					NumberOfClasses = 0.0;
```
Comment (janvanrijn) [inline src/main/java/org/openml/webapplication/fantail/dc/statistical/SimpleMetaFeatures.java:105]: I still wonder whether it wouldn't be better to strictly typecast all meta-features to int or double. For this PR it's probably fine.

File: src/main/java/org/openml/webapplication/algorithm/InstancesHelper.java (hunk 1 above)
Comment (joaquinvanschoren) [reply src/main/java/org/openml/webapplication/algorithm/InstancesHelper.java]: I previously got a compile error on getJSONArray(), but now it is fine. Probably I was using the wrong version of the api connector.

File: src/test/java/webapplication/TestFantail.java (hunk 2 above)
Comment (joaquinvanschoren) [reply src/test/java/webapplication/TestFantail.java]: I got an error like '0.0 is not equal to 0.0'. This fixed it. Just 0.0 gives you a float I guess?

File: src/main/java/org/openml/webapplication/fantail/dc/statistical/SimpleMetaFeatures.java (hunk 3 above)
Comment (joaquinvanschoren) [reply src/main/java/org/openml/webapplication/fantail/dc/statistical/SimpleMetaFeatures.java:105]: sounds ok to me

File: src/main/java/org/openml/webapplication/algorithm/InstancesHelper.java [hunk 4]
```diff
@@ -125,7 +125,7 @@ else if( total > 0.0 )
 		return result;
 	}
 	
-	public static CostMatrix doubleToCostMatrix(JSONArray cm) {
+	public static CostMatrix doubleToCostMatrix(JSONArray cm){
```
Comment (janvanrijn) [inline src/main/java/org/openml/webapplication/algorithm/InstancesHelper.java]: Please leave a space between function signature and accolade

File: src/test/java/webapplication/TestFantail.java (hunk 2 above)
Comment (janvanrijn) [reply src/test/java/webapplication/TestFantail.java]: Of course, makes sense.

Comment (janvanrijn) [reply src/test/java/webapplication/TestFantail.java]: Please add a space between the two fn arguments

**Review by janvanrijn — APPROVED** [review]
